misc/history/src/utils.ts

```diff
@@ -0,0 +1,68 @@
+import { RegisterPayload, GetMessagesPayload, GetMessagesResponse } from "./types";
+import fetch from "isomorphic-unfetch";
+import type { ICore } from "@walletconnect/types";
+
+export const DEFAULT_HISTORY_URL = "https://history.walletconnect.com";
+
+export const DAY_IN_MS = 86400 * 1000;
+
+export class HistoricalMessages {
+  public constructor(private core: ICore, public messageResponse: GetMessagesResponse) {}
+
+  public injectIntoRelayer() {
+    for (const message in this.messageResponse.messages) {
+      this.core.relayer.provider.events.emit("payload", message);
+    }
+  }
+}
+
+export class HistoryClient {
+  private jwt = "";
+
+  public constructor(private core: ICore, private readonly historyUrl = DEFAULT_HISTORY_URL) {}
+
+  public async registerTags(payload: RegisterPayload, historyUrl = DEFAULT_HISTORY_URL) {
+    try {
+      await fetch(`${historyUrl}/register`, {
+        method: "POST",
+        body: JSON.stringify(payload),
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${await this.getJwt()}`
+        },
+      });
+
+    } catch (e) {
+      throw new Error("Failed to register tags");
```

Comment:
As outlined below, just a suggestion here, there may be a more effective way to handle + forward the error in this case:

```suggestion
      throw new Error(`[HistoryClient] Failed to register tags: ${JSON.stringify(e)}`);
```